### PR TITLE
Changing schedule logic, removing status as its not needed

### PR DESCRIPTION
**Pull request by TrevorJTClarke**

### CODE REVIEW

**Comment by TrevorJTClarke** [comment]
Fixes #27

File: manager/src/lib.rs [hunk 1]
```diff
@@ -497,10 +499,11 @@ impl CronManager {
         // Remove task from schedule
         // Get previous task hashes in slot, find index of task hash, remove
         let next_slot = self.get_slot_from_cadence(task.cadence.clone());
-        let mut slot_slots = self.slots.get(&next_slot).unwrap_or(Vec::new());
```
Comment (mikedotexe) [inline manager/src/lib.rs:500]: didn't care for slot_slots? 🤪

File: manager/src/lib.rs [hunk 2]
```diff
@@ -550,58 +551,35 @@ impl CronManager {
 
         // After popping, ensure state is rewritten back
         if using_floor_key {
-            self.slots.insert(&self.slots.floor_key(&current_slot).unwrap(), &slot_data);
```
Comment (mikedotexe) [inline manager/src/lib.rs:553]: good catch, I didn't notice this

File: manager/src/lib.rs [hunk 3]
```diff
@@ -550,58 +551,35 @@ impl CronManager {
...
+            self.slots.insert(&slot_ballpark.unwrap(), &slot_data);
         } else {
             self.slots.insert(&current_slot, &slot_data);
         }
 
+        // Clean up slot if no more data
+        if slot_data.len() < 1 { self.slots.remove(&slot_ballpark.unwrap()); }
```
Comment (mikedotexe) [inline manager/src/lib.rs:560]: here it is 💥

File: manager/src/lib.rs [hunk 4]
```diff
@@ -625,33 +602,60 @@ impl CronManager {
...
+                    base64::encode(task_hash.clone())
+                );
             }
             PromiseResult::NotReady => unreachable!(),
         };
 
-        // TODO: NOTE: Is there a time when this task should be deleted?
-        self.tasks.insert(&task_hash, &task);
+        // only skip scheduling if user didnt intend
+        let current_slot = self.get_slot_id(None);
```
Comment (mikedotexe) [inline manager/src/lib.rs:640]: awesome, thanks for putting this logic in the callback

**Review by mikedotexe — APPROVED** [review]

File: manager/src/lib.rs (hunk 1 above)
Comment (TrevorJTClarke) [reply manager/src/lib.rs:500]: haha

File: manager/src/lib.rs (hunk 2 above)
Comment (TrevorJTClarke) [reply manager/src/lib.rs:553]: it was a little weird

**Pull request merged**